### PR TITLE
OKTA-378564 ASP.NET MVC > Add strong name / signature to the Asp.Net …

### PR DESCRIPTION
**Pull request by andriizhegurov-okta**
…library

Strong names added to Asp.Net assembly and also to Okta.AspNet.Abstractions as it's referenced from Asp.Net and Okta.AspNet.Test because of `InternalsVisibleTo` Attribute used.

## Code

- [ ] Unit test(s)
- [ ] Implementation

## Additional Context

This is a first stage of adding strong names. 
The Assemblies are only partially signed and should be added to ignore list on developers' machines.
Only public key is included.

Full strong names are to be added on a build machine. 

* For dev machines following commands need to be executed to enable debugging of the partially signed assemblies:
```
sn.exe -Vr (path-to)\Okta.AspNet.dll
sn.exe -Vr (path-to)\Okta.AspNet.Abstractions.dll
sn.exe -Vr (path-to)\Okta.AspNet.Test.dll
```
Also the commands above should be executed on a build machine if we're running tests from Okta.AspNet.Test. Another option is to run these tests after the assemblies are fully signed as it's shown below.

* On Jenkins build machine:
Install key pair from `okta.aspnet.pfx` into a key container named for example `OktaDotnetStrongname` (password to the file will be asked):
```
sn -i (path-to-keypair)\okta.aspnet.pfx OktaDotnetStrongname
```

Apply full strong name before the NuGet package is built:
```
sn -Rc (path-to-buildArtifacts)\Okta.AspNet.dll OktaDotnetStrongname
sn -Rc (path-to-buildArtifacts)\Okta.AspNet.Abstractions.dll OktaDotnetStrongname
sn -Rc (path-to-buildArtifacts)\Okta.AspNet.Test.dll OktaDotnetStrongname
```

### CODE REVIEW

File: Okta.AspNet/AssemblyInfo.cs [hunk 1]
```diff
@@ -5,4 +5,8 @@
 
 using System.Runtime.CompilerServices;
 
-[assembly: InternalsVisibleTo("Okta.AspNet.Test")]
+[assembly: InternalsVisibleTo("Okta.AspNet.Test, PublicKey=0024000004800000940000000602000000240000525341310004000001000100c9a1a6c8fc5450"+
```
Comment (bryanapellanes-okta) [inline Okta.AspNet/AssemblyInfo.cs:8]: How is the publickey value here deterined?

**Review by bryanapellanes-okta — APPROVED** [review]
Minor question, approved 👍

File: Okta.AspNet/AssemblyInfo.cs (hunk 1 above)
Comment (andriizhegurov-okta) [reply Okta.AspNet/AssemblyInfo.cs:8]: it's a dump of a public part of the key pair, it's added to the PR (`okta.aspnet.pk`).

Comment (andriizhegurov-okta) [reply Okta.AspNet/AssemblyInfo.cs:8]: I got it using sn.exe

File: build.cake [hunk 2]
```diff
@@ -142,6 +163,7 @@ Task("Default")
     .IsDependentOn("Clean")
     .IsDependentOn("Restore")
     .IsDependentOn("Build")
+	.IsDependentOn("Strongname")
```
Comment (laura-rodriguez) [inline build.cake:166]: Nit: Fix indentation

**Review by laura-rodriguez — APPROVED** [review]
🎉

**Pull request merged**